oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/converter/ObjectToListStringConverter.java

```diff
@@ -59,7 +59,7 @@ public Object convert(Object source, TypeDescriptor sourceType, TypeDescriptor t
 		if (source instanceof List) {
 			List<?> sourceList = (List<?>) source;
 			if (!sourceList.isEmpty() && sourceList.get(0) instanceof String) {
-				return source;
+				return new ArrayList<String>((List<? extends String>) sourceList);
```

Comment:
What happens if the source is a `List<Object>` where the first element is a `String` but the others are not? This probably applies to the `ObjectToMapStringObjectConverter` as well.